rfcs/document_all_media_properties/20230307-project_proposal.md

```diff
@@ -0,0 +1,147 @@
+
+- A clear definition of what the property represents.
+- An explanation of where the property is sourced from in the catalog, and any
+  special considerations that should be taken into account when extracting it.
+  For instance, in the case of the `creator` field, we will specify whether to
+  use the `author`, `photographer`, or both.
+- The property's data type, and whether it is required or nullable. We will also
+  indicate if the property is required but not present in some data, such as
+  `license_url` in the `meta_data` field. If the property has a set of distinct
+  values, we will note where they are in the code.
+- Whether the property is served by the API, and if so, on which endpoints
+  (search or single result, or both). We will also highlight any differences
+  between the property's type in the API versus the catalog, as noted in the
+  database, Elasticsearch, or API documentation.
+- If the API does not take the data directly from the catalog database, we will
+  indicate how the data is computed, and where this occurs (during data refresh,
+  indexing, or as a fallback in the API). We will provide a link to the relevant
+  code, if available.
+- We will also document whether the property is used by the frontend, whether
+  the type matches the catalog/API response type, and whether the frontend
+  performs any cleanup or sanitization of the property. A link to the relevant
+  code will be provided, if available.
+
+#### Tools
+
+The main tool to use for this table is an online spreadsheet app. I'm a little
+uncertain of whether the amount of information would be too much to fit into one
+table. If it is, it might be a good idea to split the data into a table with
+more standard pieces of information, and a list of all data properties with
+longer descriptions and things like code links and the criteria for selecting
+data from provider APIs.
+
+#### The process for updating the table
+
+In the future, when the data is cleaned or changed in any way, we will need to
+update the table accordingly. If we keep it in the documentation site, we can
+use PRs to keep the history of changes, linking to all the related PRs.
+
+#### Localization concerns
+
+We should identify whether any pieces of data are translated or left as is, and
+whether it might be necessary (and possible) to translate any pieces. There is
+an [open issue](https://github.com/WordPress/openverse/issues/589) about
+translating audio genres, for example.
+
+#### Limitations for `meta_data` field
+
+It will not be possible to document all the fields inside the `meta_data`
+column. It is a JSON column that was used to collect "whatever doesn't fit in
+the other fields" and its values can vary greatly. It would be extremely time-
+and resources- consuming to try to extract \_all_available meta_data fields. For
+this column, we will describe what we currently expect as a bare minimum value
+for it, and also the meta_data fields that are used in the provider scripts
```

Comment:
This is unfortunate and I think we should correct going forward giving recommendation of what to store here. In my mind it was designated for the popularity metrics.